### PR TITLE
Use early exporter

### LINKED ISSUE #11: Editorial changes
1.  Caption of Figure 2: 

> TLS Client Providing Evidence to TLS Server.

replace by TLS Server Providing Evidence to TLS Client.

2. The draft is using * throughout as a bullet. Please use something else for bullet as * has a specific meaning in the context of TLS. According to Figure 1 in [TLS draft](https://datatracker.ietf.org/doc/draft-ietf-tls-rfc8446bis/09/)/[RFC](https://datatracker.ietf.org/doc/html/rfc8446):

> *Indicates optional or situation-dependent messages/extensions that are not always sent.

### LINKED ISSUE #14: Bullets
The Intro sentence "Several formats for encoding evidence are available..." should use a bulleted list, instead of inline dashes.

### PR DESCRIPTION
**Pull request by ionut-arm**
Change the channel binding to use early exporter + editorial changes.

Fixes #11 
Fixes #14

### CODE REVIEW

File: draft-fossati-tls-attestation.md [hunk 1]
```diff
@@ -218,18 +219,18 @@ of operation when used with TLS, namely:
 - TLS server is the attester, and
 - TLS client and server mutually attest towards each other. 
 
-We will show the message exchanges of the three cases in
-sub-sections below.
+We will show the message exchanges of the first two cases in sub-sections
+below. The last case will be described in future drafts.
```
Comment (thomas-fossati) [inline draft-fossati-tls-attestation.md:223]: We can be more precise:
```suggestion
We will show the message exchanges of the first two cases in {{attester-cli}} and {{attester-srv}}.
The last case is obtained by composing the first two.
```

This needs adding anchors to the referenced sections like this:
```
## TLS Client as Attester {#attester-cli}
```
and
```
## TLS Server as Attester {#attester-srv}
```

File: draft-fossati-tls-attestation.md [hunk 2]
```diff
@@ -456,12 +457,15 @@ attestation_channel_binder = {
 ~~~~
 {: #figure-tls-binder title="Format of TLS channel binder."}
 
-* Nonce is the value provided as a challenge by the relying party.
-* The identity key public fingerprint (ik_pub_fingerprint) is a hash of the
+- Nonce is the value provided as a challenge by the relying party.
+- The identity key public fingerprint (ik_pub_fingerprint) is a hash of the
```
Comment (thomas-fossati) [inline draft-fossati-tls-attestation.md:461]: I reckon _identity key public fingerprint_ can be trimmed down to _identity key fingerprint_ without loss of precision -- and with a small but sensible increase in readability :-)

```suggestion
- The identity key fingerprint (ik_pub_fingerprint) is a hash of the
```

File: draft-fossati-tls-attestation.md [hunk 3]
```diff
@@ -456,12 +457,15 @@ attestation_channel_binder = {
...
   Subject Public Key Info from the leaf X.509 certificate transmitted in
   the handshake.
-* The channel binder (channel_binder) is a partial transcript of the TLS
-  handshake, up to (but not including) the Certificate message.
+- The channel binder (channel_binder) is a value obtained from the early
+  exporter mechanism offered by the TLS implementation (see section 7.5 of
+  {{RFC8446}}). This Early Exporter Value (EEV) must be obtained immediately
```
Comment (thomas-fossati) [inline draft-fossati-tls-attestation.md:466]:
```suggestion
  exporter mechanism offered by the TLS implementation ({{Section 7.5 of
  RFC8446}}). This Early Exporter Value (EEV) must be obtained immediately
```

**Review by thomas-fossati — COMMENTED** [review]
thanks.  I've left a few editorial nits

File: draft-fossati-tls-attestation.md [hunk 4]
```diff
@@ -218,18 +219,18 @@ of operation when used with TLS, namely:
...
-service can feed the nonce into the generation of the evidence. The TLS 
-server, when receiving the evidence, will have to contact the verifier 
-(which is not shown in the diagram). 
+In this use case, the TLS server (acting as a relying party) challenges the TLS
+client (as the attester) to provide evidence. The TLS client is the attester
+and the TLS server acts as a relying party. The TLS server needs to provide a
+nonce in the EncryptedExtensions message to the TLS client so that the
+attestation service can feed the nonce into the generation of the evidence. The
+TLS server, when receiving the evidence, will have to contact the verifier
+(which is not shown in the diagram).
```
Comment (yaronf) [inline draft-fossati-tls-attestation.md:233]: The 2nd sentence of this paragraph duplicates the first one.

File: draft-fossati-tls-attestation.md [hunk 5]
```diff
@@ -456,12 +457,15 @@ attestation_channel_binder = {
...
+- The identity key public fingerprint (ik_pub_fingerprint) is a hash of the
   Subject Public Key Info from the leaf X.509 certificate transmitted in
   the handshake.
-* The channel binder (channel_binder) is a partial transcript of the TLS
-  handshake, up to (but not including) the Certificate message.
+- The channel binder (channel_binder) is a value obtained from the early
+  exporter mechanism offered by the TLS implementation (see section 7.5 of
+  {{RFC8446}}). This Early Exporter Value (EEV) must be obtained immediately
+  following the ServerHello message, using 'attestation-binder' as the label
+  and an empty context.
```
Comment (yaronf) [inline draft-fossati-tls-attestation.md:468]: I think we should be more formal here, using the notation of RFC 8446, Sec. 7.5.

**Review by yaronf — CHANGES_REQUESTED** [review]

File: draft-fossati-tls-attestation.md (hunk 5 above)
Comment (ionut-arm) [reply draft-fossati-tls-attestation.md]: Alright, I've tweaked this bulletpoint a bit (thanks for the suggestion, I realised I hadn't nailed down the binder length) and added a figure to show how the call should look like. Let me know if that seems in line with what you expected!

Comment (yaronf) [reply draft-fossati-tls-attestation.md]: @ionut-arm Yep, this is much better. I just think that the context is an empty string (""), not an empty list ([]).

**Review by hannestschofenig — APPROVED** [review]

**Pull request merged**